src/lambdas/ingest/index.js

```diff
@@ -53,6 +53,7 @@ export const handler = async (event, _context) => {
 
   if (event.create_indices) {
     await createIndex('collections')
+    return
```

Comment:
When the initial "collection" index is created, we shouldn't process the message any further. If we do, we get a failure and and error message that it's not an Item or Collection.

Reply:
I will certainly defer to your expertise here. I went back to the ingest handler ing the previous version (`1.1.0`) and I don't see an early return

https://github.com/stac-utils/stac-server/blob/v1.1.0/src/lambdas/ingest/index.js#L55

Are we losing functionality by returning early that exited pre 2.x? Should we have been exiting early all along?

Reply:
Interesting – would some kind of delay allow it to work? (Or do we not want to get into sketchy hacks)

Reply:
I believe we should have had an early return all along.  In the previous code, that message would have gotten to here:

```
      if (data && data.hasOwnProperty('extent')) {
        index = COLLECTIONS_INDEX
      } else if (data && data.hasOwnProperty('geometry')) {
        index = data.collection
      } else {
        // @ts-ignore
        next()
        return
      }
```

and returned with no error. Now it goes through:

```
export async function convertIngestObjectToDbObject(
  // eslint-disable-next-line max-len
  /** @type {{ hasOwnProperty: (arg0: string) => any; type: string, collection: string; links: any[]; id: any; }} */ data
) {
  let index = ''
  logger.debug('data', data)
  if (isCollection(data)) {
    index = COLLECTIONS_INDEX
  } else if (isItem(data)) {
    index = data.collection
  } else {
    throw new InvalidIngestError(
      `Expeccted data.type to be "Collection" or "Feature" not ${data.type}`
    )
  }
```

And results in an error if it's not a Feature or Collection. These are the only 3 types of message the ingest handles, so that's all of the cases covered.

Reply:
@MattBialas we don't actually want to sent a notification for this out on the SNS topic. This is an initialization message that creates the `collections` index. Without this early return, the message then goes through the regular processing as if it were a Feature or Collection message, and throws an error that it's not either of those.

Reply:
Thanks for clarifying 👍